SHARE-336 Flavors in Featured and Examples

FelixAuf opened the pull request:
- Changed the flavor strings in featured and example programs to reference the new flavor entity
- The migration converts the old strings to the new reference
- Adjusted admin area, homepage and tests to work with these changes
- Added more tests to admin/example_programs (now similar to featured)
- On the homepage, the "Featured Programs" header is now only visible if there actually are featured programs (can revert that change easily)

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [x] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Verify that all tests are passing, if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [ ] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### Additional Description
`TODO: Add additional information that is not in your commit-message here`

### Tests - additional information
`TODO: add additional information about testruns here`

dmetzner commented on assets/js/custom/Index.js:
```diff
@@ -29,10 +29,11 @@ function Index (clickStats, homepageClickStats, confirmButtonText) {
 
       /* eslint-disable no-undef */
-      if (flavor !== undefined && flavor !== 'pocketcode') {
+      if ((flavor !== undefined && flavor !== 'pocketcode') || category === 'example') {
```
The flavor should not be undefined, not even when the category is "example". Modify the parentheses.

dmetzner reviewed (changes requested):
Rest looks good to me!

FelixAuf committed:
SHARE-336 Flavors in Featured and Examples

- Changed the flavor strings in featured and example programs to reference the new flavor entity
- The migration converts the old strings to the new reference
- Adjusted admin area, homepage and tests to work with these changes
- Added more tests to admin/example_programs (now similar to featured)
- On the homepage, the "Featured Programs" header is now only visible if there actually are featured programs (can revert that change easily)
